hail/python/hailtop/config/deploy_config.py

```diff
@@ -104,6 +110,25 @@ async def get_healthcheck(request):  # pylint: disable=unused-argument,unused-va
 
         return root_app
 
+    async def addresses(self, service: str) -> List[Tuple[str, int]]:
+        from ..auth import service_auth_headers  # pylint: disable=cyclic-import
+        namespace = self.service_ns(service)
+        headers = service_auth_headers(self, namespace)
+        async with ssl_client_session(
+                raise_for_status=True,
+                timeout=aiohttp.ClientTimeout(total=5),
+                headers=headers) as session:
+            async with await session.get(
+                    self.url('address', f'/api/{namespace}/{service}')) as resp:
+                dicts = await resp.json()
+                return [(d['address'], d['port']) for d in dicts]
+
+    async def address(self, service: str) -> Tuple[str, int]:
+        service_addresses = await self.addresses(service)
+        n = len(service_addresses)
+        assert n > 0
```

Comment:
Should there be a nicer error message than this? It will throw a 500.

Reply:
This won't usually be used inside a web server. It's for clients to discover an IP address for a container providing a hail service, such as `shuffler`.

I'm not really sure what should happen if we ask for an address and there are none available (for example, if all the containers happen to be down). We should probably retry waiting for the service to come alive. I'm inclined to leave it like this until we have more experience using address.